Xext/vidmode.c

```diff
@@ -408,13 +410,25 @@ ProcVidModeGetAllModeLines(ClientPtr client)
     if (!pVidMode->GetFirstModeline(pScreen, &mode, &dotClock))
         return BadValue;
 
-    int payload_len = modecount * ((ver < 2) ? sizeof(xXF86OldVidModeModeInfo)
-                                             : sizeof(xXF86VidModeModeInfo));
+    x_rpcbuf_t rpcbuf = { .swapped = client->swapped };
+
+    Bool ok = TRUE;
+    do {
+        if (ver < 2)
+            ok &= fillModeInfoV1(&rpcbuf, dotClock, mode);
+        else
+            ok &= fillModeInfoV2(&rpcbuf, dotClock, mode);
+    } while (pVidMode->GetNextModeline(pScreen, &mode, &dotClock));
+
+    if (!ok) {
+        x_rpcbuf_clear(&rpcbuf);
+        return BadAlloc;
+    }
 
     xXF86VidModeGetAllModeLinesReply rep = {
         .type = X_Reply,
         .length = bytes_to_int32(sizeof(xXF86VidModeGetAllModeLinesReply) -
-                                 sizeof(xGenericReply) + payload_len),
+                                 sizeof(xGenericReply) + rpcbuf.wpos),
```

Comment:
I got confused when I looked at this and realized that there is a bug in the old code, first caused by this: 877b18d4580c0b7aff9a55aae0c9fcaea50d9d98. Here is what is looked like before: https://github.com/X11Libre/xserver/blob/de636d491bad83a6e67d404928a6c82b628300cf/Xext/vidmode.c#L359 . This new code fixes that automatically by not calculating length manually.